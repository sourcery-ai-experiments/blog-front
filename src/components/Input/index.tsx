"use client";

import { useMemo, type InputHTMLAttributes } from "react";
import { useFormStatus } from "react-dom";

const Input = ({
  errors,
  className,
  label,
  ...rest
}: InputHTMLAttributes<HTMLInputElement> & {
  errors?: string[];
  className?: string;
  label?: string;
}) => {
  const { pending } = useFormStatus();

  const id = useMemo(() => {
    return rest.id ?? `${rest.name}_input`;
  }, [rest.id, rest.name]);

  return (
    <div className="flex flex-col gap-2">
      {label && <label htmlFor={id}>{label}</label>}
      <input
        id={id}
        className={
          "w-full rounded border-none px-4 py-3.5 text-sm leading-5 ring-1 ring-inset " +
          className
        }
        disabled={pending}
        {...rest}
      />
      {errors &&
        errors.map((error, index) => (
          <span key={index} className="text-sm text-red-500">
            {error}
          </span>
        ))}
    </div>
  );
};

export default Input;
