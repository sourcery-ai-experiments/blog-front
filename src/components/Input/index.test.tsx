import { render } from "@testing-library/react";
import Input from "@/components/Input";

jest.mock("react-dom", () => ({
  ...jest.requireActual("react-dom"),
  useFormStatus: () => ({ pending: false }),
}));

describe("Input", () => {
  it("snapshot test", () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });
});

describe("Input", () => {
  it("snapshot test", () => {
    const { container } = render(<Input name="test" label="Test" />);

    expect(container).toMatchSnapshot();
  });
});
