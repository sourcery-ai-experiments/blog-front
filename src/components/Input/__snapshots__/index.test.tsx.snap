// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot test 1`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <input
      class="w-full rounded border-none px-4 py-3.5 text-sm leading-5 ring-1 ring-inset undefined"
      id="undefined_input"
    />
  </div>
</div>
`;

exports[`Input snapshot test 2`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <label
      for="test_input"
    >
      Test
    </label>
    <input
      class="w-full rounded border-none px-4 py-3.5 text-sm leading-5 ring-1 ring-inset undefined"
      id="test_input"
      name="test"
    />
  </div>
</div>
`;
